grubby.c

```diff
@@ -1913,6 +1913,14 @@ static int writeConfig(struct grubConfig *cfg, char *outName,
                         return 1;
                 }
 
+		int outtempfd = open(tmpOutName, O_RDONLY);
```

Comment:
This looks really weird, and I dont think it'll result in what you're hoping for (two different, unrelated file descriptors). To call `fsync()` on the `FILE*` you can obtain a file descriptor with `fileno()`. Doing this before the `fclose()` above makes it better understandable.